tools/topology/topology2/platform/intel/sdw-jack-generic.conf

```diff
@@ -186,56 +186,103 @@ Object.Pipeline.host-gateway-capture [
 		Object.Widget.host-copier.1 {
 			stream_name	"Passthrough Capture 0"
 			pcm_id 1
-			num_input_audio_formats 9
+			num_input_audio_formats 3
 			num_output_audio_formats 9
-			Object.Base.audio_format.1 {
-				# 32/32 -> 16/16 bits conversion is done here
-				in_bit_depth	32
-				in_valid_bit_depth	32
-			}
-			# 32-bit 48KHz 1ch
-			Object.Base.input_audio_format.7 {
-				in_channels		1
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				in_ch_cfg	$CHANNEL_CONFIG_MONO
-				in_ch_map	$CHANNEL_MAP_MONO
-			}
-			Object.Base.output_audio_format.7 {
-				out_bit_depth		32
-				out_valid_bit_depth	32
-				out_channels		1
-				out_ch_cfg	$CHANNEL_CONFIG_MONO
-				out_ch_map	$CHANNEL_MAP_MONO
-			}
-			# 24-bit 48KHz 1ch
-			Object.Base.input_audio_format.8 {
-				in_channels		1
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				in_ch_cfg	$CHANNEL_CONFIG_MONO
-				in_ch_map	$CHANNEL_MAP_MONO
-			}
-			Object.Base.output_audio_format.8 {
-				out_bit_depth		32
-				out_valid_bit_depth	24
-				out_channels		1
-				out_ch_cfg	$CHANNEL_CONFIG_MONO
-				out_ch_map	$CHANNEL_MAP_MONO
-			}
-			# 16-bit 48KHz 1ch
-			Object.Base.input_audio_format.9 {
-				in_channels		1
-				in_bit_depth		32
-				in_valid_bit_depth	32
-				in_ch_cfg	$CHANNEL_CONFIG_MONO
-				in_ch_map	$CHANNEL_MAP_MONO
-			}
-			Object.Base.output_audio_format.9 {
-				out_channels		1
-				out_ch_cfg	$CHANNEL_CONFIG_MONO
-				out_ch_map	$CHANNEL_MAP_MONO
-			}
+			Object.Base.input_audio_format [
+				# 32-bit 48KHz 2ch
+				{
+					in_bit_depth 32
+					in_valid_bit_depth 32
+				}
+				# 32-bit 48KHz 4ch
+				{
+					in_bit_depth 32
+					in_valid_bit_depth 32
+					in_channels 4
+					in_ch_cfg	$CHANNEL_CONFIG_3_POINT_1
+					in_ch_map	$CHANNEL_MAP_3_POINT_1
+				}
+				# 32-bit 48KHz 1ch
+				{
+					in_channels		1
+					in_bit_depth		32
+					in_valid_bit_depth	32
+					in_ch_cfg	$CHANNEL_CONFIG_MONO
+					in_ch_map	$CHANNEL_MAP_MONO
+				}
+			]
+
+			CombineArrays.Object.Base.output_audio_format [
+				# array of 2ch formats with range of out valid bit depths
+				{
+					out_bit_depth [
+						32
+					]
+					out_valid_bit_depth [
+						24
+						32
+					]
+				}
+				# array of 4ch formats with range of out valid bit depths
+				{
+					out_bit_depth [
+						32
+					]
+					out_valid_bit_depth [
+						24
+						32
+					]
+					out_channels [
+						4
+					]
+					out_ch_cfg [
+						$CHANNEL_CONFIG_3_POINT_1
+					]
+					out_ch_map [
+						$CHANNEL_MAP_3_POINT_1
+					]
+				}
+				# array of 1ch formats with range of out valid bit depths
+				{
+					out_bit_depth [
+						32
+					]
+					out_valid_bit_depth [
+						24
+						32
+					]
+					out_channels [
+						1
+					]
+					out_ch_cfg [
+						$CHANNEL_CONFIG_MONO
+					]
+					out_ch_map [
+						$CHANNEL_MAP_MONO
+					]
+				}
+				# 16-bit 2ch
+				{
+					out_bit_depth		16
```

Comment:
is 
````
out_bit_depth		16
````
the same as 
````
out_bit_depth	[
	16
]
````
if yes, why not always use the former case for single-value configurations?

Reply:
this is a feature or a limitation with the way the CombineArrays feature has been defined in the compilet today. So all values even single values have to be specified with an array. I've updated the patch to use array consistenly now